### PR TITLE
OCP metadata library

### PR DESCRIPTION
**Pull request by rsevilla87**
### Description

Adding a library to gather OCP metadata.
Apart from the above, it also implements a couple of helper functions to retrieve the cluster's Prometheus info and default ingress domain

- Fixes: #3

### CODE REVIEW

**Comment by vishnuchalla** [comment]
Looks good to me. Just one request, Can we move all the private methods to the bottom of the file. I would want to maintain this organisation throughout this library for better organisation and clarity.

**Comment by vishnuchalla** [comment]
Good to go from my side. @jtaleric please let me know your thoughts on this. Thanks!

File: ocp-metadata/ocp-metadata.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+			// obtain the version from the last completed update
+			versionInfo.ocpVersion = update.Version
+			break
+		}
+	}
+	return versionInfo, err
+}
+
+// getNodesInfo returns node information
+func (meta *Metadata) getNodesInfo() (nodeInfo, error) {
```
Comment (jtaleric) [inline ocp-metadata/ocp-metadata.go:208]: Should we add the region each node is in?

File: ocp-metadata/ocp-metadata.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+// GetClusterMetadata returns a clusterMetadata object from the given OCP cluster
+func (meta *Metadata) GetClusterMetadata() (ClusterMetadata, error) {
+	metadata := ClusterMetadata{MetricName: metadataMetricName}
+	infra, err := meta.getInfraDetails()
+	if err != nil {
+		return metadata, nil
+	}
+	metadata.ClusterName, metadata.Platform = infra.Status.InfrastructureName, infra.Status.Platform
+	for _, v := range infra.Status.PlatformStatus.Aws.ResourceTags {
```
Comment (jtaleric) [inline ocp-metadata/ocp-metadata.go:58]: what happens here w/ non AWS?

File: ocp-metadata/ocp-metadata.go (hunk 1 above)
Comment (jtaleric) [reply ocp-metadata/ocp-metadata.go:208]: We seem to capture this with the infraObj.

File: ocp-metadata/ocp-metadata.go (hunk 2 above)
Comment (rsevilla87) [reply ocp-metadata/ocp-metadata.go:58]: good point, I'll add a conditional

File: ocp-metadata/ocp-metadata.go (hunk 1 above)
Comment (rsevilla87) [reply ocp-metadata/ocp-metadata.go:208]: Done, at the moment it only captures the AWS region, seems like the infrastructure object from Azure or GCP does not provide that info. We'll have to implement different alternatives in the future (getting it from the node labels, machine set etc...)

**Review by jtaleric — APPROVED** [review]
lgtm - we just need to document (maybe in a follow on PR) that the metadata is pretty specific to AWS in its current state.

**Comment by vishnuchalla** [comment]
> lgtm - we just need to document (maybe in a follow on PR) that the metadata is pretty specific to AWS in its current state.

@rsevilla87 if there are some parts of it specific to AWS, can we add small comments over there just for the user's reference within this PR, please. For extending its implementation for different platforms can be taken care in different PRs.

**Pull request merged**